.travis.yml

```diff
@@ -1,21 +1,19 @@
 language: node_js
-addons:
-  code_climate:
-    repo_token: CODE_CLIMATE_TOKEN
+os:
+- linux
+- osx
+- windows
 node_js:
-- '11'
-- '10'
+- '13' # Changing this? Remind to adjust the linter condition below
```

Comment:
> What does this comment mean?

@jiahaog see a few lines below; with the addition of {Windows, macOS} platforms in Travis, CI duration lengthened, so I added a condition restricting the linter to run only on a single sub-build, to shave off 10s * 8 from the build:
```yaml
# Only run linter once, for faster CI
- if [ "$TRAVIS_OS_NAME" = "linux" ] && [ "$TRAVIS_NODE_VERSION" = "13" ]; then npm run lint; fi
```

I just pushed a little improvement of the comment to clearer `# Changing this? Remind to adjust the linter condition below causing linter to run for only one version (for faster CI)`

---

> Thank you so so much for taking the time to do this!! I ran some basic workflows on a mac and it works great as well.

@jiahaog welcome 🙂, and thanks for the final tests. I use lots of open-source software at home and at work, this is for me an occasion to give back in a context that I know (I do TS at work these days), and contribute a lot by cleaning & strengthening a codebase that needed it. The last few weeks I got confident that at least common use cases will be fine, as:

1. I got a fair amount of feedback in the last weeks in the PR, as well as in a few existing issues where I suggested reporters to give this branch a try because it fixes their issues. Feedback was largely positive, and a few minor remaining bugs were reported and fixed.
2. The added tests, added CI runs for all platforms, scripting of a pre-publish manual test, and final exploratory testing on Linux & Windows, are all reassuring.

There will inevitably be regressions given the amount of changes, but I'm not anticipating a tornado of bug reports. I'm subscribed to issues notifications and will keep a close eye on them after releasing, fixing regressions quickly in rapid patch versions.

> Please let me know if you need any help merging / publishing this! Also, will you be squashing the commits when you do the merge?

@jiahaog I'll be merging as 8.0.0 this today or next week, with the description of this PR as release notes. I don't anticipate needing help merging & publishing, but will ping you if something I forgot about goes wrong.

Thanks for the support and reviews, cheers!